File: src/main/java/com/example/airmed/Entity/Psychotherapist.java
package com.example.airmed.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Psychotherapist {
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false, unique = true)
    private String medicalNumber;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String mail;

    @Column(nullable = false, unique = true)
    private String phone;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String country;

    @Column(nullable = false)
    private String locality;

    @Column
    private String cabinetLocation;

    @Column
    private String linkLocation;

    @Column(nullable = false)
    private int priceConsult;

    @Column(nullable = false)
    private int priceConsultation;

    @Column(columnDefinition = "boolean default false")
    private boolean online;

    @Column(columnDefinition = "boolean default false")
    private boolean CNAS;
}

File: frontend/src/classes/Psychotherapist.ts
export class Psychotherapist{
    id: number;
    medicalNumber: string;
    firstName: string;
    lastName: string;
    mail: string;
    phone: string;
    bio: string;
    country: string;
    locality: string;
    cabinetLocation: string;
    linkLocation: string;
    priceConsult: number;
    priceConsultation: number;
    online: boolean;
    CNAS: boolean;

    constructor(id: number,
                medicalNumber: string,
                firstName: string,
                lastName: string,
                mail: string,
                bio: string,
                phone: string,
                country: string,
                locality: string,
                cabinetLocation: string,
                linkLocation: string,
                priceConsult: number,
                priceConsultation: number,
                online: boolean,
                CNAS: boolean) {
        this.id = id;
        this.priceConsultation = priceConsultation;
        this.priceConsult = priceConsult;
        this.CNAS = CNAS;
        this.online = online;
        this.bio = bio;
        this.medicalNumber = medicalNumber;
        this.firstName = firstName;
        this.lastName = lastName;
        this.mail = mail;
        this.phone = phone;
        this.country = country;
        this.locality = locality;
        this.cabinetLocation = cabinetLocation;
        this.linkLocation = linkLocation;
    }

    static jsonToPsychotherapist(json: string) {
        const data = JSON.parse(json);
        // console.log("Data ", data);
        return new Psychotherapist(
            data.id,
            data.medicalNumber,
            data.firstName,
            data.lastName,
            data.mail,
            data.bio,
            data.phone,
            data.country,
            data.locality,
            data.cabinetLocation,
            data.linkLocation,
            data.priceConsult,
            data.priceConsultation,
            data.online,
            data.CNAS
        );
    }
}

export default function psychotherapistToJson(psychotherapist: Psychotherapist): string {
    const json = JSON.stringify({
        id: psychotherapist.id,
        medicalNumber: psychotherapist.medicalNumber,
        firstName: psychotherapist.firstName,
        lastName: psychotherapist.lastName,
        bio: psychotherapist.bio,
        mail: psychotherapist.mail,
        phone: psychotherapist.phone,
        country: psychotherapist.country,
        locality: psychotherapist.locality,
        cabinetLocation: psychotherapist.cabinetLocation,
        linkLocation: psychotherapist.linkLocation,
        priceConsult: psychotherapist.priceConsult,
        priceConsultation: psychotherapist.priceConsultation,
        online: psychotherapist.online,
        CNAS: psychotherapist.CNAS
    });
    return json;
}